File: problems/2224_minimum_number_of_operations_to_convert_time/solution.py
class Solution:
    def convertTime(self, current: str, correct: str) -> int:
        def toMinutes(t: str) -> int:
            h, m = map(int, t.split(':'))
            return h * 60 + m
        diff = toMinutes(correct) - toMinutes(current)
        ops = 0
        for d in [60, 15, 5, 1]:
            ops += diff // d
            diff %= d
        return ops

File: problems/2224_minimum_number_of_operations_to_convert_time/Solution.java
class Solution {
    public int convertTime(String current, String correct) {
        int diff = toMin(correct) - toMin(current);
        int ops = 0;
        for (int d : new int[]{60, 15, 5, 1}) {
            ops += diff / d;
            diff %= d;
        }
        return ops;
    }
    private int toMin(String t) {
        return Integer.parseInt(t.substring(0, 2)) * 60 + Integer.parseInt(t.substring(3));
    }
}
